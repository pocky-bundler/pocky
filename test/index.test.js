import path from 'path'
import Bundler from '../src'

const hideProjectPath = file => {
  return {
    ...file,
    path: file.path.replace(path.join(__dirname, '../'), '/$ownDir/'),
    assets: file.assets.map(asset => hideProjectPath(asset))
  }
}

test('assets', async () => {
  const input = path.join(__dirname, 'fixtures/simple/input.js')
  const bundler = new Bundler(input)
  await bundler.bundle()
  expect(hideProjectPath(bundler.tree)).toMatchSnapshot()
})
