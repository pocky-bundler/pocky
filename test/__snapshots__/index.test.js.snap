// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assets 1`] = `
Object {
  "assets": Array [
    Object {
      "assets": Array [],
      "content": "export default 'foo'
",
      "importNames": Array [
        Object {
          "imported": "default",
          "local": "foo",
        },
      ],
      "importPath": "./foo",
      "path": "/$ownDir/test/fixtures/simple/foo",
    },
    Object {
      "assets": Array [],
      "content": "export default 'bar'
",
      "importNames": Array [
        Object {
          "imported": "bar",
          "local": "bar",
        },
      ],
      "importPath": "./nest/bar",
      "path": "/$ownDir/test/fixtures/simple/nest/bar",
    },
  ],
  "content": "import foo from './foo'
import { bar } from './nest/bar'
",
  "isRoot": true,
  "path": "/$ownDir/test/fixtures/simple/input.js",
}
`;
